pkg/syncer/cnsoperator/controller/cnsregistervolume/cnsregistervolume_controller.go

```diff
@@ -201,6 +201,11 @@ func (r *ReconcileCnsRegisterVolume) Reconcile(request reconcile.Request) (recon
 	log.Debugf("CNS Volume create spec is: %+v", createSpec)
 	vol, err := r.volumeManager.CreateVolume(ctx, createSpec)
 	if err != nil {
+		if instance.Spec.DiskURLPath != "" {
```

Comment:
If VolumeID is FCD and is already registered with CNS, CNS throws CnsFault. CNS throws CNSFault for every other issues which it can't handle. So there is no way of differentiating it if volume ID is FCD ID.

But when DiskURLPath is used, CNS throws AlreadyRegistered fault which is already handled in volumeManager.CreateVolume. So for any other error, we can fail the operation.

Reply:
We have this taken care at - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/cns-lib/volume/manager.go#L217-L223

so as per this code, if we get CnsAlreadyRegisteredFault, it will be logged and operation is successful, but for other faults, we are returning an error.

Reply:
Yes exactly.

Reply:
I mean all other errors will be returned, so we need to set them - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/cns-lib/volume/manager.go#L233-L235

Reply:
Can you at least log an error message, if the server is not returning fault with which we can distinguish between volume is already registered or other CNS fault is received for registration using `CnsRegisterVolumeSpec->VolumeID`?

Some notes will be helpful for future reference.

Reply:
Done!